src/LuaShip.cpp

```diff
@@ -243,6 +282,33 @@ static int l_ship_set_skin(lua_State *l)
 }
 
 /*
+ * Method: SetPattern
+ *
+ * Changes the pattern used for texturing the ship.
+ *
+ * > ship:SetPattern(num)
+ *
+ * Parameters:
+ *
+ *   num - the pattern number
+ *
+ * Example:
+ *
+ * > ship:SetPattern(5)
+ *
+ * Status:
+ *
+ *  experimental
+ */
+static int l_ship_set_pattern(lua_State *l)
+{
+	Ship *s = LuaObject<Ship>::CheckFromLua(1);
+	unsigned int num = lua_tointeger(l, 2);
```

Comment:
@laarmen - okay, it takes into consideration that indexing starts at 0 now (amended my last commit).

Along these lines - wouldn't the following line in Ship.cpp need fixing then too?:
`GetModel()->SetPattern(Pi::rng.Int32(0, GetModel()->GetNumPatterns()));`

Or am I showing my C++ ignorance here?